tests/unit/amo/components/TestCategories.js

```diff
@@ -45,26 +46,56 @@ describe('Categories', () => {
       dispatch: sinon.stub(),
     };
 
-    return findDOMNode(findRenderedComponentWithType(renderIntoDocument(
+    return findRenderedComponentWithType(renderIntoDocument(
       <Provider store={store}>
         <CategoriesBase i18n={getFakeI18nInst()} {...baseProps} {...props} />
       </Provider>
-    ), CategoriesBase));
+    ), CategoriesBase);
   }
 
-  it('renders Categories', () => {
+  function renderDomNode(props) {
+    return findDOMNode(render(props));
+  }
+
+  it('dispatches setCurrentView with addonType', () => {
+    const fakeDispatch = sinon.stub();
     const root = render({
-      addonType: 'extension',
+      addonType: ADDON_TYPE_EXTENSION,
+      dispatch: fakeDispatch,
+    });
+
+    expect(fakeDispatch.calledWith(setCurrentView({
+      addonType: ADDON_TYPE_EXTENSION,
+    }))).toBeTruthy();
+
+    // Make sure that we update the addonType when `componentDidUpdate()`
+    // is called. This will happen when changing from one route that uses
+    // this component to another–the props will be updated so
+    // `componentDidUpdate()` is called without the component being
+    // mounted again.
+    // TODO: This feels naughty; can it be done better?
```

Comment:
Yeah I think Enzyme will give us better hooks into component life-cycle when using `mount`. But I'll look at that when we start converting things over.